### DISCUSSION TITLE
How to reset branch to other brach?

### DISCUSSION THREAD

**Discussion by azzamsa** [NeogitOrg/neogit#1139] [Q&A] [answered]
Hi, 👋

In Magit I can do:
1) Press X to launch the reset menu
2) Press s to select soft reset
3) Select desired branch (*or* tag *or* commit hash)

How to do step 3 in Neogit? (Select the desired branch)

These are the steps I need to take in Magit:

![2024-01-18_10-37](https://github.com/NeogitOrg/neogit/assets/17734314/823c2104-0635-4b4e-9757-301d86fc6eee)

![2024-01-18_10-37_1](https://github.com/NeogitOrg/neogit/assets/17734314/84cadba3-4dac-45f9-b4ef-5bf60ebf5a11)

**Comment by azzamsa** [comment]
Hi @CKolkey, what's your workflow for `resetting to another branch`? I always resort to using Magit for this.

**Reply by CKolkey** [reply]
`bX` gives you a picker to choose - is that what you mean? On nightly its also available via `Xb`

**Reply by azzamsa** [reply]
> gives you a picker to choose 

But it doesn't give me a choice between `soft` and `hard` reset?

**Reply by CKolkey** [reply]
oh, true. Does magit?

**Reply by azzamsa** [reply]
Yes, as I attached the screenshot in my original question.

**Reply by CKolkey** [reply, marked as answer]
right, sorry, github just showed me the thread, not the parent, for some reason. 

https://github.com/NeogitOrg/neogit/assets/7228095/2029d9c6-9b4d-4507-95f3-e5770fdd288d

It's working as you describe on `nightly` - that'll hit `master` when nvim 0.10 is released :)

**Reply by azzamsa** [reply]
Wow, that fast.
Thank you so much 🍕

**Reply by CKolkey** [reply]
Haha, no worries! And, to be fair, I made this change a few months ago.. nightly has about 4+ months of work waiting for nvim 0.10 😅
